### ISSUE TITLE
Lift 14-character limit on version context names

### ISSUE THREAD

**Issue by dadlerj** [sourcegraph/sourcegraph#11591]
Requested by https://app.hubspot.com/contacts/2762526/company/768958891

**Comment by rrhyne** [comment]
The easiest solution is to increase the limit it to 50 characters. That will eat into search bar real-estate, but it's the org's choice. Using ... or similar to shorten the label shouldn't be done because no matter where we put the ... it could obscure key information about the version being searched.

**Comment by poojaj-tech** [comment]
@dadlerj has this been an ask from other companies as well? I'm curious to know how many companies/teams have longer version context names and will potentially see the shorter search bar UI.

**Comment by dadlerj** [comment]
I think we only have a small handful of companies using version contexts right now given they're still experimental. That said, the other one that I did a live setup call with definitely was a bit confused when they saw the error message, and ended up just removing letters from the words they were planning to use (making the version name look a bit janky). I can't say confidently that it's a broad issue yet though.

**Comment by poojaj-tech** [comment]
@dadlerj was the error message about crossing the 14 character limit? Thanks for the context! We should solve for the jankiness before the feature launches. I think having a consistent base Search bar UI for the core features (interactive/plain-text options, amount of search real estate space, structural/regex options) across companies is important for understanding and improving usability at scale.

**Comment by dadlerj** [comment]
@poojaj-tech yep. The error message didn't explain why we limited it (it was just a default GraphQL error indicating that the strong was too long).

The "jankiness" was more on the customer side, since they wanted to find a good name for the context that was <=14 chars so they could get rid of the error. Just removing characters from words (I can't recall exactly what it was, but, e.g., if we created an internal context that was like "srcgraph 3.14" to get under the limit, that would be equivalent to what they did.

And yeah, the reasons for limiting make total sense — agreed this deserves some design thinking!
